tools/check-upcoming-maintenance.py

```diff
@@ -0,0 +1,130 @@
+import os
+import re
+import subprocess
+from typing import List
+from optparse import OptionParser
+
+class NodeMaintenance:
+    '''
+    Class to keep track of project, zones, and search terms for nodes, as well as results from gcloud queries
+    '''
+    def __init__(self, project, zones, regex):
+        self.project = project
+        res = subprocess.run("gcloud projects describe {}".format(self.project), shell=True, capture_output=True, text=True)
+        if res.returncode != 0:
+            raise Exception("{} does not exist or you may not have permission to access it".format(self.project))
+        
+        self.zones = None
+        if zones:
+            self.zones = zones.split(",")
+            for zone in self.zones:
+                res = subprocess.run("gcloud compute zones describe {}".format(zone), shell=True, capture_output=True, text=True)
+                if res.returncode != 0:
+                    raise Exception("{} does not exist".format(zone))
+        
+        self.regex = None
+        self.screen_reader = False
+        res = subprocess.run("gcloud config get accessibility/screen_reader", 
+                             shell=True, capture_output=True, text=True)
+        if res.returncode != 0:
+            raise Exception("Error getting accessibility/screen_reader information")
+        if res.stdout.split("\n")[0] == "True":
+            self.screen_reader = True
+        if regex:
+            self.regex = re.compile(regex)
+        self.maint_nodes = None
+        self.upc_maint = None
```

Comment:
Is there any reason to make it a part of state ?

Reply:
Done.  Let me know if you had something else in mind.

Reply:
why not just pass `maint_nodes` and `upc_maint` to `__init__` , while calling ` get_upcoming_maintenance` and `get_maintenance_nodes` from `node_maintenace_factory`?